core/src/main/java/hudson/TcpSlaveAgentListener.java

```diff
@@ -415,7 +415,9 @@ public boolean connect(Socket socket) throws IOException {
                             LOGGER.log(Level.FINE, "Expected ping response from {0} of {1} got {2}", new Object[]{
                                     socket.getRemoteSocketAddress(),
                                     new String(ping, "UTF-8"),
-                                    new String(response, 0, responseLength, "UTF-8")
+                                    responseLength > 0 && responseLength <= response.length ?
```

Comment:
FTR this fix is for `UsageStatisticsTest` last failure:

![image](https://user-images.githubusercontent.com/223853/49905076-a7cdb700-fe6c-11e8-84f6-323f098fa87a.png)